README.md

```diff
@@ -99,18 +115,28 @@ A request for execution should contain the following request parameters:
 * `operationName` - [*Optional*]: The name of the Operation in the Document to execute.
 * `variables` - [*Optional*]: Values for any Variables defined by the Operation.
 
+Note: Be aware that `query` is a misleading name as it can contain a string describing the operation that may be either a query, a mutation or a subscription.
+
 Note: depending on the serialization format used, values of the aforementioned parameters can be
 encoded differently but their names and semantics must stay the same.
 
 Note: specifying `null` in JSON (or equivalent values in other formats) as values for optional request parameters is equivalent to not specifying them at all.
 
 ## GET
 
-For HTTP GET requests, the query parameters should be provided in the query component of the request URL in the form of
+For HTTP GET requests, the query parameters MUST be provided in the query component of the request URL in the form of
 `key=value` pairs with `&` symbol as a separator and both the key and value should have their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
-The unencoded value of the `variables` parameter should be represented as a JSON-encoded string.
+The unencoded value of the `variables` parameter MUST be represented as a URL-encoded JSON string.
```

Comment:
```suggestion
The value of the `variables` parameter MUST be represented as a URL-encoded JSON object, with a name-value pair for each variable and its value.
```

This is what the example shows: the unencoded/decoded value is a JSON object, the actual/encoded value is a URL-encoded JSON object. No double-encoding.

Reply:
nit pick (not blocking): JSON is a string (the `N` is for notation).

Reply:
I'm still confused by `The unencoded value ... represented as a URL-encoded ...`.

Does this mean that I 
- first have to translate the variables object into JSON, 
- then URL-encode that JSON to get an _un_-encoded parameter value, 
- and then again percent-encode the _un_-encoded parameter value before sending it in the HTTP request?

I hope percent/URL-encoding happens only once.